tests/integration/test_karapace.py

```diff
@@ -15,8 +14,15 @@
 import socket
 
 
+@contextmanager
+def allocate_port_no_reuse() -> Iterator[int]:
```

Comment:
**[nit-pick]**
Maybe `Generator[int]` as the return type, since you already did earlier, but doesn't hurt as generators as also iterators.